src/common/core/librarycore.js

```diff
@@ -0,0 +1,785 @@
+                        collKey = key.substr(0, key.length - CONSTANTS.COLLECTION_NAME_SUFFIX.length);
+                        for (i = 0; i < selfOverlays[key].length; i += 1) {
+                            removeRelationFromOverlay(overlay, collKey, selfOverlays[key][i], path);
+                        }
+                    } else {
+                        removeRelationFromOverlay(overlay, key, path, selfOverlays[key]);
+                    }
+                }
+            }
+
+            function moveLibraryRelations(root, from, to) {
+                var overlay = self.getChild(root, CONSTANTS.OVERLAYS_PROPERTY),
+                    fromOverlay = JSON.parse(JSON.stringify(self.getProperty(overlay, from))),
+                    tempOverlay,
+                    index,
+                    key, collKey, i;
+
+                delete fromOverlay._mutable;
+                for (key in fromOverlay) {
+                    if (key.indexOf(CONSTANTS.COLLECTION_NAME_SUFFIX) !== -1) {
+                        collKey = key.substr(0, key.length - CONSTANTS.COLLECTION_NAME_SUFFIX.length);
+                        for (i = 0; i < fromOverlay[key].length; i += 1) {
+                            tempOverlay = JSON.parse(JSON.stringify(self.getProperty(overlay, fromOverlay[key][i])));
+                            delete tempOverlay._mutable;
+                            tempOverlay[collKey] = to;
+                            self.setProperty(overlay, fromOverlay[key][i], tempOverlay);
+                        }
+                    } else {
+                        tempOverlay = JSON.parse(JSON.stringify(self.getProperty(overlay, fromOverlay[key])));
+                        index = tempOverlay[key + CONSTANTS.COLLECTION_NAME_SUFFIX].indexOf(from);
+                        if (index) {
+                            tempOverlay[key + CONSTANTS.COLLECTION_NAME_SUFFIX] = to;
+                            delete tempOverlay._mutable;
+                            self.setProperty(overlay, fromOverlay[key], tempOverlay);
+                        }
+                    }
+                }
+
+                self.deleteProperty(overlay, from);
+                self.setProperty(overlay, to, fromOverlay);
+            }
+
+            //</editor-fold>
+
+            //<editor-fold=Modified Methods>
+            this.loadRoot = function (hash) {
+                return TASYNC.call(function (root) {
+                    root.libraryRoots = getLibraryRootsInfo(root);
+                    return root;
+                }, innerCore.loadRoot(hash));
+            };
+
+            this.createNode = function (parameters) {
+                var node;
+
+                if (parameters && parameters.parent &&
+                    (self.isLibraryRoot(parameters.parent) || self.isLibraryElement(parameters.parent))) {
+                    return new Error('cannot create new node inside a library');
```

Comment:
Start sentence with upper case and end it with a dot.
"cannot" could mean that it could do it under other conditions -> "Not allowed to ..."